[Chen Hsiao Ting] iP

hsiaotingluv opened the pull request:
# DukePro
> “Your mind is for having ideas, not holding them.” – David Allen [source](https://dansilvestre.com/productivity-quotes)

DukePro frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* FAST SUPER FAST to use

All you need to do is,
1. download it from [here.](https://nus-cs2103-ay2122s1.github.io/website/schedule/week4/project.html)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

-----------------------------------------

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

hsiaotingluv committed:
Update Duke.java

Level-1

hsiaotingluv committed:
Level-2, Add, List

hsiaotingluv committed:
Level 3-Mark as Done

hsiaotingluv committed:
Level-4-ToDo-Event-Deadline

hsiaotingluv committed:
Level-6-Delete

hsiaotingluv committed:
Documentation

hsiaotingluv committed:
branching

hsiaotingluv committed:
testing

hsiaotingluv committed:
Merge branch 'master' into feature1

hsiaotingluv committed:
feature 1

hsiaotingluv committed:
master

hsiaotingluv committed:
Merge branch 'feature1'

hsiaotingluv committed:
undo testing

hsiaotingluv committed:
testing

hsiaotingluv committed:
testing

hsiaotingluv committed:
Add-Automated-Text-Ui-Testing

hsiaotingluv committed:
Level-7-Save

hsiaotingluv committed:
Merge branch 'Level-7-Save'

# Conflicts:
#	src/main/java/Duke.java

hsiaotingluv committed:
Level-7-Save

hsiaotingluv committed:
Level-7-Save

hsiaotingluv committed:
Level-7-Save

hsiaotingluv committed:
Level-7-Save

hsiaotingluv committed:
Level-7-Save

hsiaotingluv committed:
Reformat main function

hsiaotingluv committed:
Added date and time

hsiaotingluv committed:
Added more OOP

hsiaotingluv committed:
Added A-Packages

hsiaotingluv committed:
Adds Java Doc

hsiaotingluv committed:
Adds Coding Standard

hsiaotingluv committed:
Add Level 9 Find

hsiaotingluv committed:
resolved merge conflicts

hsiaotingluv committed:
merge branch-Level-9 to master

hsiaotingluv committed:
Add A-JUnit.

hsiaotingluv committed:
Add A-JUnit.

hsiaotingluv committed:
Add manifest.

yuifuku1118 commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,97 @@
+    /**
+     * Returns the type of task.
+     *
+     * @return string type of task.
```
I like how you named your method names with verbs! (Easy to understand at one look)

yuifuku1118 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,109 @@
+package duke;
+
+import tasks.Deadline;
+import tasks.Event;
+import tasks.Task;
+import tasks.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import java.util.ArrayList;
+import java.util.Scanner;
```
It good that you separate imports by package. Nice and well organized!

hsiaotingluv committed:
Add build gradle

yuifuku1118 commented on src/main/java/tasks/Event.java:
```diff
@@ -0,0 +1,43 @@
+        String duration = splitted[1].trim();
+
+        // split start date and end date
+        String[] splitted2 = duration.split("/to ", 2);
```
Maybe you could change the variable names splittted and splitted2 to something more meaningful?
(Sorry that its a really small thing, this was like the only thing that i could find as a possible improvement)

yuifuku1118 reviewed (commented):
Sorry saw an allocation wrongly pls ignore this;(

hsiaotingluv committed:
Add check style

hsiaotingluv committed:
Add GUI

hsiaotingluv committed:
Create GUI and interact with user

hsiaotingluv committed:
Use FXML

aakanshanarain reviewed (commented):
Looks good! All methods and variables are easy to read and proper coding standards are followed throughout! Couldn't really find any error in the coding style!

timothywongej commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,181 @@
+    }
+
+    @Override
+    public void start(Stage stage) {
```
Some recently committed methods should have a Javadoc header comment if they're public

timothywongej commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,58 @@
+                    System.out.println("Bye. Hope to see you again soon!");
+                } else if (str.equals("list")) {
+                    System.out.println(tasks.list());
+                } else if (str.contains("find")) {
```
Would using ```contain``` result in an error if the user inputs something such as ```test find test```?

timothywongej commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,97 @@
+     *
+     * @return a string representation of the deleted task.
+     */
+    public String delete() {
```
The naming convention for this method is slightly odd - ```delete``` returns a String. May I suggest something like```getDeletedTask```?

timothywongej commented on src/main/java/tasks/TaskList.java:
```diff
@@ -0,0 +1,145 @@
+     * @param str user input to indicate a task is done.
+     * @return description of task that is marked done.
+     */
+    public String done(String str) {
```
Perhaps consider changing this method name to a verb as well

timothywongej commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,97 @@
+    /**
+     * Sets the task to done by marking its status.
+     */
+    public void done() {
```
May want to differentiate the name of this method and the one below it for clarity.

timothywongej reviewed (commented):
Well done! I think you did a great job. 👍 🥇 

Perhaps look into the flagged naming conventions for some methods. (verbs are recommended!)

hsiaotingluv committed:
Added OOP

hsiaotingluv committed:
Added documentations

hsiaotingluv committed:
Added checkstyle

hsiaotingluv committed:
Added new JAR file

hsiaotingluv committed:
Added new JAR file

hsiaotingluv committed:
Assertion: Use assert feature

Assert documents important assumptions that should hold at various points in the code.

Assertions used to define assumptions about the program state so that the runtime can verify them.

Let's add an assertion to verify assumption about the postcondition of getIndex method.

hsiaotingluv committed:
Code Quality: improve code quality

Ui included in several command classes. Some codes have differing format.

Refactor class Ui to include all interactions with user. Reformat the code to increase readability.

Let's,
* extract all the interaction with user and include them inside the class Ui
* reformat all the codes to improve code quality

hsiaotingluv committed:
Merge pull request #1 from hsiaotingluv/branch-A-Assertions

Assertion: Use assert feature

hsiaotingluv committed:
Merged branch-A-Assertions

hsiaotingluv committed:
Merge branch 'master' of github.com:hsiaotingluv/ip

hsiaotingluv committed:
Merge pull request #2 from hsiaotingluv/branch-A-CodeQuality

Code Quality: improve code quality

hsiaotingluv committed:
Merged branch-A-CodeQuality

hsiaotingluv committed:
Resolved merge conflict

hsiaotingluv committed:
Using Stream: use Streams feature of Java in code.

Storage of data is only done after user enter bye and exit the chat bot.

Changes made to update storage file immediately after user enter a valid command.
This saves the data even if user quit the program without entering 'bye'.

Let's use the streams feature in storage class to store the data with short and concise codes.

hsiaotingluv committed:
Merge branch 'master' into branch-A-Streams

hsiaotingluv committed:
Merge pull request #3 from hsiaotingluv/branch-A-Streams

Using Stream: use Streams feature of Java in code.

hsiaotingluv committed:
Added Continuous Integration

hsiaotingluv committed:
Added C-Extension Help

hsiaotingluv committed:
Merge pull request #5 from hsiaotingluv/branch-C-Help

Added C-Extension Help

hsiaotingluv committed:
Delete build

hsiaotingluv committed:
Add checkstyle

hsiaotingluv committed:
Add checkstyle

hsiaotingluv committed:
Merge branch 'master' of github.com:hsiaotingluv/ip

hsiaotingluv committed:
Update README.md

hsiaotingluv committed:
Improve GUI

hsiaotingluv committed:
Improve GUI

hsiaotingluv committed:
Add checkStyle

hsiaotingluv committed:
Add User Guide

hsiaotingluv committed:
Update README.md

hsiaotingluv committed:
Update Ui photo

hsiaotingluv committed:
Delete Ui.png

hsiaotingluv committed:
Update Ui photo

hsiaotingluv committed:
Update README.md

hsiaotingluv committed:
Set theme jekyll-theme-hacker

hsiaotingluv committed:
Set theme jekyll-theme-hacker

hsiaotingluv committed:
Better Ui

hsiaotingluv committed:
Merge branch 'master' of github.com:hsiaotingluv/ip

hsiaotingluv committed:
Set theme jekyll-theme-slate

hsiaotingluv committed:
Update README.md

hsiaotingluv committed:
Update README.md

hsiaotingluv committed:
Update README.md

hsiaotingluv committed:
Add Jar

hsiaotingluv committed:
Update JAR file

hsiaotingluv committed:
Update README.md

hsiaotingluv committed:
Update README.md

hsiaotingluv committed:
Update README.md

hsiaotingluv committed:
Improve error exception catching

hsiaotingluv committed:
Resolve deadline and event bug

hsiaotingluv committed:
Improve GUI

hsiaotingluv committed:
Update jar link
